articles/what-is-jenkins/index.md

```diff
@@ -0,0 +1,61 @@
+
+### Problems Developers faced before Continuous Integration
+Earlier before Continuous Integration was introduced, a team of developers used to work from different locations, each of them would individually implement a few classes. They would write code and tests to ensure that their classes work properly. Then assume the software project will build and run smoothly because classes are tested well. They then integrate the classes and combine them to create a larger program that was being developed by the team, but suddenly everything breaks.  There are issues with integration, the code fails to compile, all sorts of bugs arise, and things stop working out. This is also called [integration hell](https://www.solutionsiq.com/agile-glossary/integration-hell/#:~:text=Integration%20Hell%20refers%20to%20the,that%20it%20can%20finally%20integrate.)
+Integration hell is a major problem to software projects and it is risky to their success reason. Seeing it was hard to estimate how long it would take to fix the things that went wrong. Developers had to figure out what went wrong, and that could be very hard to do in a large project. After diagnosing the problems, again time had to be spent fixing them, the project budget may also increase and the success of the project is risked. The solution to integration hell was to integrate continuously throughout the entire project, and thus the practice of Continuous Integration was introduced to the DevOps field.
+
+### What is Continuous Integration?
+CI is a development practice whereby developers are required to commit changes to the source code in a shared repository more frequently. CI says developers must integrate code early and often, so every time changes are pushed to the repository (such as GitHub) the project code should be checked into, compiled, and tested. The productivity of developers has been improved by Continuous Integration, rather than before when they used to wait until all components are developed. Then spend weeks and months working on bugs and errors that would [inevitably arise](https://www.edureka.co/blog/what-is-jenkins/).
+
+Continuous Integration allows the development teams to detect problems early on. It also automates the process of compiling, testing, and reporting the project source code. When a developer push changes into a repository, the CI server gets notified about the changes. The CI server either pulls the repository after a few minutes to see if there are any changes or the repository actively calls the CI server to let it know when there are changes. It then clones the repository and builds the project. If the project fails it notifies the team. The continuous integration server finally generates various reports such as the line coverage of the project test or style issues found on the [project code](https://www.edureka.co/blog/what-is-jenkins/).
```

Comment:
Check your grammar and for run-on sentences. You can use an extension like grammarly.

"It then clones..." What does? The server? Be specific.